vault/resource_consul_secret_backend_role.go

```diff
@@ -165,13 +170,22 @@ func consulSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) erro
 	nodeIdentities := d.Get("node_identities").(*schema.Set).List()
 
 	data := map[string]interface{}{
-		"policies":           policies,
-		"consul_policies":    policies,
 		"consul_roles":       roles,
 		"service_identities": serviceIdentities,
 		"node_identities":    nodeIdentities,
 	}
 
+	checkConsulPolicies, _, err := semver.GreaterThanOrEqual(ctx, client, consts.VaultVersion11)
```

Comment:
nit: the `checkConsulPolicies` variable name seems to imply that we are checking some policies but it seems to only be used for selecting which field to use when configuring the consul policies. Wonder if we could come up with a more generic variable name.  The way that the semver check is implemented makes it tricky since we have to handle the error each time.

Reply:
I tentatively put `useAPIVer2` to match the below suggestion.